test/test_torch.py

```diff
@@ -951,6 +951,45 @@ def test_where_bool_tensor(self):
             res = torch.where(a > 0)
             self.assertEqual(1, len(res))
 
+    def test_where_tensor_cpu(self):
+        def rand_tensor(size, dtype, device, cont):
+            if dtype.is_floating_point:
+                return torch.rand(size=size, dtype=dtype, device=device)
+            elif dtype == torch.uint8:
+                return torch.randint(1, 5, size=size, dtype=dtype, device=device)
+            elif dtype == torch.bool:
+                return torch.randint(0, 1, size=size, dtype=dtype, device=device).bool()
+            else:
+                return torch.randint(-5, 5, size=size, dtype=dtype, device=device)
+
+        def get_tensor(size, dtype, device, cont):
+            if not cont and len(size) < 2:
+                raise RuntimeError("Unable to generate non contiguous tensor with size < 2")
+            t = rand_tensor(size, dtype, device, cont)
+            if cont:
+                return t
+            else:
+                return t.transpose(0, 1)
+
+        height = 5
+        width = 5
+        for device in torch.testing.get_all_device_types():
+            for dt1 in torch.testing.get_all_math_dtypes(device):
+                for dt2 in torch.testing.get_all_math_dtypes(device):
+                    for cont in [True, False]:
+                        x1 = get_tensor((height, width), dt1, device, cont)
+                        x2 = get_tensor((height, width), dt2, device, cont)
+                        if dt1 != dt2:
+                            self.assertRaisesRegex(RuntimeError, "expected scalar type", lambda: torch.where(x1 == 1, x1, x2))
+                        else:
+                            res = torch.where(x1 == 1, x1, x2)
```

Comment:
wait -- is this going to very rarely happen?  Like, never if x1 is floating point?

Reply:
> how about a cosine_embedding_loss and kl_div test?

Added